server-spi-private/src/main/java/org/keycloak/models/utils/KeycloakModelUtils.java

```diff
@@ -763,6 +764,38 @@ private static String[] formatPathSegments(String[] segments, int index, String
         return segments;
     }
 
+    public static GroupModel findGroupByPath(GroupProvider groupProvider, RealmModel realm, String path) {
```

Comment:
Is there some reason to keep the old `findGroupByPath` implementation? Can we replace that method everywhere?

Reply:
Backwards compatibility?

Reply:
Good question I'm not sure to replace the method everywhere in the first step.

Reply:
I don't think we need to keep backward compatibility for the util method in the `server-spi-private` module. Let's keep this as is currently and do a follow-up where we remove the old method later. I created an issue tracker for it: https://github.com/keycloak/keycloak/issues/23344

Reply:
+1 to handle this in separate issue